src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntTests.cs

```diff
@@ -608,7 +622,25 @@ .maxstack  1
             getArgs(builder, sourceType: "nuint", destType: "nint?", expectedImplicitIL: null, expectedExplicitIL: convToNullableT("conv.i", "nint"), expectedCheckedIL: convToNullableT("conv.ovf.i.un", "nint"));
             getArgs(builder, sourceType: "float", destType: "nint?", expectedImplicitIL: null, expectedExplicitIL: convToNullableT("conv.i", "nint"), expectedCheckedIL: convToNullableT("conv.ovf.i", "nint"));
             getArgs(builder, sourceType: "double", destType: "nint?", expectedImplicitIL: null, expectedExplicitIL: convToNullableT("conv.i", "nint"), expectedCheckedIL: convToNullableT("conv.ovf.i", "nint"));
-            getArgs(builder, sourceType: "decimal", destType: "nint?", expectedImplicitIL: null, expectedExplicitIL: "...");
+            getArgs(builder, sourceType: "decimal", destType: "nint?", expectedImplicitIL: null,
+@"{
+  // Code size       13 (0xd)
+  .maxstack  1
+  IL_0000:  ldarg.0
+  IL_0001:  call       ""long decimal.op_Explicit(decimal)""
+  IL_0006:  conv.i
+  IL_0007:  newobj     ""nint?..ctor(nint)""
```

Comment:
>nint [](start = 25, length = 4)

Can we change the way this is displayed?  `nint` doesn't exist at the IL level.

Reply:
When generating IL, we generate a set of tokens for the untranslated symbols referenced in IL (see `CommonPEModuleBuilder.GetFakeSymbolTokenForIL`). Those fake tokens are then translated later when emitting (see `MetadataWriter.ResolveEntityHandleFromPseudoToken` which eventually calls `PEModuleBuilder.Translate`).

The IL visualizer in the test code (see `ILBuilderVisualizer.ILBuilderToString`) is handed the untranslated symbols and those are the symbols displayed. The visualizer has similar behavior for references to `dynamic`.

---
In reply to: [357284988](https://github.com/dotnet/roslyn/pull/40271#discussion_r357284988) [](ancestors = 357284988)

Reply:
[VisualizeRealIL](http://sourceroslyn.io/#Roslyn.Test.Utilities/CommonTestBase.cs,597ddb4eb3fb6b98) does that.